src/main/java/core/basesyntax/Rectangle.java

```diff
@@ -0,0 +1,27 @@
+package core.basesyntax;
+
+public class Rectangle extends Figure {
+    private final double firstLeg;
+    private final double secondLeg;
+    private final ColorSupplier color;
+
+    public Rectangle(ColorSupplier color, double firstLeg, double secondLeg) {
```

Comment:
The constructor should not take `ColorSupplier` as a parameter. Instead, pass the `Color` type directly.